stdlib/Test/src/Test.jl

```diff
@@ -311,7 +311,7 @@ function eval_test(evaluated::Expr, quoted::Expr, source::LineNumberNode, negate
 
     Returned(res,
```

Comment:
Same on line 308, this expects `res` to be a Bool

Reply:
Line 308 doesn’t expect `res` to be a `Bool`, it just expects a type for which a `Base.:!` is defined. Otherwise it errors, like it should. In my opinion, that’s the desired behavior (if we don’t want code inside `@test` to behave differently from code outside of it).